controllers/dogu_controller_test.go

```diff
@@ -0,0 +1,64 @@
+package controllers
+
+import (
+	"context"
+	k8sv1 "github.com/cloudogu/k8s-dogu-operator/api/v1"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"testing"
+	"time"
+)
+
+func Test_evaluateRequiredOperation(t *testing.T) {
+
+	logger := log.FromContext(context.TODO())
+
+	t.Run("installed should return upgrade", func(t *testing.T) {
+		ldapCr.Status = k8sv1.DoguStatus{Status: k8sv1.DoguStatusInstalled}
+
+		operation, err := evaluateRequiredOperation(ldapCr, logger)
+		require.NoError(t, err)
```

Comment:
These tests are concise an readable. I suggest to separate the function-under-test-call from following tests though for even more readability. For small test functions this does not state a real problem. But the larger a test function grows the less readability it will gain.

In other words, `NoError()` is a test step and belongs to the `Equal()` test below.